### PR TITLE
added pg:diagnose

### PR DESCRIPTION
**Pull request by jdx**

### CODE REVIEW

File: commands/diagnose.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      body: params
+    })
+  })
+
+  let displayReport = report => {
+    cli.log(`Report ${report.id} for ${report.app}::${report.database}
+available for one month after creation on ${report.created_at}
+`)
+    let display = checks => {
+      if (!checks.length) return
```
Comment (msakrejda) [inline commands/diagnose.js]: Is this necessary? If `checks` is empty, the code won't execute the body of the loop and exit, right?

File: commands/diagnose.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    display(report.checks.filter(c => c.status === 'red'))
+    display(report.checks.filter(c => c.status === 'yellow'))
+    display(report.checks.filter(c => c.status === 'green'))
+    display(report.checks.filter(c => !['red', 'yellow', 'green'].find(d => d === c.status)))
+  }
+
+  let report
+  let id = args['DATABASE|REPORT_ID']
+  if (id && id.match(/^[a-z0-9\-]{36}$/)) {
```
Comment (msakrejda) [inline commands/diagnose.js:59]: You don't need the backslash here, I think: a trailing `-` in a matching group is not treated as a special character.

File: lib/util.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  return sortBy(keys, k => k !== 'DATABASE_URL', 'name')
+}
+
+exports.databaseNameFromUrl = (uri, config) => {
+  const url = require('url')
+
+  let names = exports.configVarNamesFromValue(config, uri)
+  let name = names.pop()
+  while (name === 'DATABASE_URL') name = names.pop()
```
Comment (msakrejda) [inline lib/util.js]: This should probably be the subject of another PR, but I noticed this behavior breaks when `DATABASE_URL` is the only attachment:

```
maciek@mothra:~$ h pg:info --app young-ridge-18903
=== 
Plan:        Hobby-dev
Status:      Available
Connections: 0/20
PG Version:  9.5.4
Created:     2016-11-01 18:56 UTC
Data Size:   7.4 MB
Tables:      0
Rows:        0/10000 (In compliance)
Fork/Follow: Unsupported
Rollback:    Unsupported
Add-on:      postgresql-infinite-56908

maciek@mothra:~$ h addons --app young-ridge-18903

Add-on                                         Plan       Price  State  
─────────────────────────────────────────────  ─────────  ─────  ───────
heroku-postgresql (postgresql-infinite-56908)  hobby-dev  free   created
 └─ as DATABASE
```

(note the empty name after the `===` header in the `pg:info` output)

**Comment by msakrejda** [comment]
+1

**Review by msakrejda — APPROVED** [review]

File: commands/diagnose.js (hunk 1 above)
Comment (jdx) [reply commands/diagnose.js]: yeah I think it was before a refactor, removed

File: commands/diagnose.js (hunk 2 above)
Comment (jdx) [reply commands/diagnose.js:59]: it is a special character, because you use it for declaring ranges of characters like `[a-zA-Z]+`

File: lib/util.js (hunk 3 above)
Comment (jdx) [reply lib/util.js]: ok so maybe this instead?

``` js
while (names.length > 0 && name === 'DATABASE_URL') name = names.pop()
```

File: commands/diagnose.js (hunk 2 above)
Comment (msakrejda) [reply commands/diagnose.js:59]: Not as the last character in a character group expression. That is,

```
> "bar-".match(/[ac-]/g)
[ 'a', '-' ]
> "bar-".match(/[a-c]/g)
[ 'b', 'a' ]
> 
```

But I don't think this matters much; it can stay as is.

File: lib/util.js (hunk 3 above)
Comment (uhoh-itsmaciek) [inline lib/util.js]: `> 1`, right? The condition is evaluated before the loop body, so you want to make sure you have something to pop off.

File: commands/diagnose.js (hunk 2 above)
Comment (jdx) [reply commands/diagnose.js:59]: ah yep, you're right
